newspapers.com.js

```diff
@@ -46,15 +46,30 @@ function detectWeb(doc, url) {
 
 function doWeb(doc, url) {
 	var newItem = new Zotero.Item("newspaperArticle");
+	var scripts = doc.getElementsByTagName("script");
+	var json = '';
+	var jsonre = /var staPageDetail = JSON.parse\("(.+?)"\);/;
+	for (var i = 0; i < scripts.length; i++) {
+		var arr = scripts[i].textContent.match(jsonre);
+		if (arr) {
+			// we have to unescape the json string this way, since javascript
+			// unescape doesn't work here
+			json = arr[1].replace(/\\\//g, "/").replace(/\\"/g, '"').replace(/\\'/g, "'");
```

Comment:
No need for this. You can just use `JSON.parse()` the same way they are. You just need to capture the double quotes in the regexp above.